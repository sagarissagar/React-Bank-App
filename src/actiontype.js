export const ADD_AMT="ADD_AMT";
export const RMV_AMT="RMV_AMT";