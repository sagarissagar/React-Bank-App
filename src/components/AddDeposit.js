import React, { useState } from 'react';

const AddDeposit = function ({  allActions }) {

  const [checktext, setCheckText] = useState('');
  const [amouttext, setAmtText] = useState('');

  const handlesetCheckText = (event) => {
    setCheckText(event.target.value);
  }

  const handlesetAmtText = (event) => {
    setAmtText(event.target.value);
  }

  const addAmount = (checktext, amouttext) => {
    allActions.depositAmount(checktext, amouttext);
    setAmtText(''); 
    setCheckText('');
    document.getElementById("myForm").reset();
  }

  return (
    <form id="myForm">
      <div className="add-deposit-tbl">
        <h4 className="text-center">Add Deposit</h4>
        <div className="row">
          <div className="col-md-6">
            <label htmlFor="add-cheque">Check Number</label>
          </div>
          <div className="col-md-6">
            <input type="number" id="add-cheque" placeholder="Add Deposit here.." onChange={handlesetCheckText}></input>
          </div>
          <div className="col-md-6">
            <label htmlFor="add-deposit">Amount</label>
          </div>
          <div className="col-md-6">
            <input type="number" id="add-deposit" onChange={handlesetAmtText} ></input>
          </div>
          <div className="btn-wrap">
            <button type="button" onClick={() => addAmount(checktext, amouttext)} disabled={!checktext || !amouttext}>Submit</button>
          </div>
        </div>
      </div>
    </form>
  )
}

export default AddDeposit;