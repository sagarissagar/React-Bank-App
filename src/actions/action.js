import { ADD_AMT, RMV_AMT } from '../actiontype';

function depositAmount(checktext, amouttext) {

  const today = new Date();
  const time = today.getHours() + ":" + today.getMinutes() + ":" + today.getSeconds();
  const date = today.toISOString().slice(0,10) + " " + time;

  return {
    type: ADD_AMT,
    date: date,
    addtype: 'credit',
    checktext,
    amouttext
  }
}

function withdrawAmount(checktext, amouttext) {

  const today = new Date();
  const time = today.getHours() + ":" + today.getMinutes() + ":" + today.getSeconds();
  const date = today.toISOString().slice(0,10) + " " + time;

  return {
    type: RMV_AMT,
    date: date,
    addtype: 'debit',
    checktext,
    amouttext
  }
}

export default { depositAmount, withdrawAmount }

