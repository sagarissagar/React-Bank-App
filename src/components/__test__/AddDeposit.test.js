import React from 'react';
import { ADD_AMT, RMV_AMT } from '../../actiontype';
import actions from '../../actions/action';
import rootReducer from '../../reducers';
import {render, fireEvent, screen, getByLabelText} from '@testing-library/react';
import AddDeposit from '../../components/AddDeposit';
import '@testing-library/jest-dom/extend-expect';

describe('actions', () => {

    const today = new Date();
    const time = today.getHours() + ":" + today.getMinutes() + ":" + today.getSeconds();
    const date = today.toISOString().slice(0, 10) + " " + time;

    // action testing
    it('add amount action', () => {
        const checktext = 100;
        const amouttext = 200;
        const expectedAction = {
            type: ADD_AMT,
            date: date,
            addtype: 'credit',
            checktext,
            amouttext
        }
        expect(actions.depositAmount(checktext, amouttext)).toEqual(expectedAction)
    })


    // reducers - for intitial test
    it('should return the initial state', () => {
        expect(rootReducer(undefined, {})).toEqual([])
    })

    // reducers - handle add amount
    it('add amountte', () => {
        const actionAdd = { 
            type: ADD_AMT,
            date: date,
            addtype: 'credit',
            checktext: 10,
            amouttext: 24,
            balance: 24
        };

        const expectedState = [{ 
            date: date, 
            type: 'credit',
            check: 10,
            amount: 24,
            balance: 24
        }];
        expect(rootReducer(undefined, actionAdd)).toEqual(expectedState);
    })

    // component testing
    test('component tsting', () => {
      const { getByLabelText } = render(<AddDeposit />);

      // By default button is disabled
      expect(screen.getByRole('button')).toBeDisabled();

      // Button enabled when both values are true
      const input = getByLabelText("Check Number");
      const input2 = getByLabelText("Amount");
      fireEvent.change(input, { target: { value: 5 } });
      fireEvent.change(input2, { target: { value: 8 } });
      expect(screen.getByRole('button')).toBeEnabled();

      // Validate label and its type
      expect(input).toHaveAttribute("type", "number");
      expect(input2).toHaveAttribute("type", "number");

    });

})


