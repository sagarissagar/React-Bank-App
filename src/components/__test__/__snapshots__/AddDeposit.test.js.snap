// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`My Connected React-Redux Component should render with given state from Redux store 1`] = `undefined`;

exports[`actions should return the initial state 1`] = `Object {}`;
